<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="viz_config_dir" default="$(find kimera_dsg_builder)/config/incremental_visualizer"/>
  <arg name="visualizer_ns" default="/kimera_dsg_visualizer"/>

  <arg name="start_rviz" default="true"/>
  <arg name="rviz_dir" default="$(find kimera_dsg_builder)/rviz"/>
  <arg name="rviz_file" default="dsg_streaming_visualizer.rviz"/>
  <arg name="rviz_path" default="$(arg rviz_dir)/$(arg rviz_file)"/>

  <arg name="verbosity" default="0"/>
  <arg name="world_frame" value="world"/>

  <include file="$(find kimera_dsg_visualizer)/launch/includes/kimera_dsg_visualizer_params.xml">
      <arg name="viz_config_dir" value="$(arg viz_config_dir)"/>
      <arg name="visualizer_ns" value="$(arg visualizer_ns)"/>
      <arg name="world_frame" default="$(arg world_frame)"/>
  </include>

  <node pkg="kimera_dsg_builder" type="dsg_visualizer_node" name="dsg_visualizer_node"
        output="screen"
        required="true"
        args="-alsologtostderr -colorlogtostderr -v=$(arg verbosity)">
    <param name="visualizer_ns" value="$(arg visualizer_ns)"/>
    <param name="load_graph" value="false"/>

    <param name="mesh_plugin_type" value="PGMO"/>
    <param name="scene_graph_filepath" value=""/>
    <param name="visualizer_ns" value="/kimera_dsg_visualizer"/>
    <param name="mesh_plugin_ns" value="dsg_mesh"/>
    <param name="output_path" value=""/>

    <remap from="~dsg" to="/incremental_dsg_builder_node/dsg"/>
    <remap from="~dsg_mesh_updates" to="/incremental_dsg_builder_node/pgmo/optimized_mesh"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" output="screen" if="$(arg start_rviz)" args="-d $(arg rviz_path)"/>

</launch>
